### ISSUE TITLE
Copy last inserted entry

### ISSUE THREAD

**Issue by ahmed-shariff** [jkitchin/org-ref#677]
Is there a functionality similar to `org-capture-goto-last-stored` or a way to store the link to the last entry? 

I apologise for the noob question.

**Comment by jkitchin** [comment]
That sounds like a good idea. What exactly did you have in mind? Right now, when you add an entry, it puts the key on the clipboard, so you can paste the key somewhere. I guess you would like to do something like C-c C-l to insert a link to the bibtex entry?

**Comment by jagrg** [comment]
If you only want to copy the last entry, you could add the last cleaned entry to the clipboard:

(add-hook 'org-ref-clean-bibtex-entry-hook
	  (lambda ()
	    (bibtex-copy-entry-as-kill)
	    (with-temp-buffer
	      (insert (car bibtex-entry-kill-ring))
	      (clipboard-kill-ring-save (point-min) (point-max))))
	  t)

Or: (add-hook 'org-ref-clean-bibtex-entry-hook 'bibtex-copy-entry-as-kill t) if you don't need the entry in the clipboard. In either case the "last" entry should be (car bibtex-entry-kill-ring), so to go to it, similar to what org-capture-goto-last-stored does, you can grab the last key and send it to bibtex-completion-show-entry like this:

(defun org-ref-goto-last-stored ()
  (interactive)
  (let (last-key
	(last-entry (car bibtex-entry-kill-ring)))
    (when last-entry
      (with-temp-buffer
	(insert last-entry)
	(bibtex-beginning-of-entry)
	(when (looking-at bibtex-entry-maybe-empty-head)
	  (setq last-key (bibtex-key-in-head))))
      (bibtex-completion-show-entry (list last-key)))))

And to insert from a list of recent entries:

(defun last-killed-entries ()
  (interactive)
  (helm :sources
	(helm-build-sync-source "Entries"
	  :candidates bibtex-entry-kill-ring
	  :multiline t
	  :action 'insert)))

To create links, you can use org-store-link (C-c l) and org-insert-link (C-c C-l) as you would normally. I haven't used this feature before, but I can tell it works with Org version 9.2.5. The syntax is: [[file:yourfile.bib::thekey]].

**Comment by ahmed-shariff** [comment]
My problem was I was using `org-brain` to connect alot of dots (with alot of research papers in the mix). I wanted to add the citation key also to the entries which would allow me to easily keep track of stuff and refer to them. So I wanted to a way to easily access the stuff I just added to the bib file. I didn't realise the key is copied to the clipboard. 

For now, what I had done is make the org file I am using to keep track the papers in `org-brain` also set as the file for notes in `org-ref`, and have the entry generated in a `org-brain` friendly way.

And thanx @jagrg, that actually gives the result I am looking for.

Closing for now.

On a different note, it can help if there is a hook after an entry has been made.
